x-pack/plugins/watcher/public/sections/watch_edit/components/threshold_watch_edit_component.tsx

```diff
@@ -150,17 +150,28 @@ const ThresholdWatchEditUi = ({
     setIndexPatterns(titles);
   };
   const loadData = async () => {
-    const theFields = await getFields(watch.index);
-    setFields(theFields);
-    setTimeFieldOptions(getTimeFieldOptions(theFields));
+    if (watch.index && watch.index.length > 0) {
+      const theFields = await getFields(watch.index);
+      setFields(theFields);
+      setTimeFieldOptions(getTimeFieldOptions(theFields));
+    }
     getIndexPatterns();
   };
   useEffect(() => {
     loadData();
   }, []);
   const { errors } = watch.validate();
   const hasErrors = !!Object.keys(errors).find(errorKey => errors[errorKey].length >= 1);
-
+  const expressionErrorMessage = i18n.translate(
+    'xpack.watcher.thresholdWatchExpression.fixErrorInExpressionBelowValidationMessage',
+    {
+      defaultMessage: 'Please fix the errors in the expression below.',
+    }
+  );
+  const expressionFields = ['termSize', 'termField', 'threshold', 'timeWindowSize'];
```

Comment:
should `aggField` be added to this as well?

<img width="718" alt="Screen Shot 2019-04-11 at 2 42 39 PM" src="https://user-images.githubusercontent.com/5226211/55982998-19294500-5c68-11e9-835d-c7941e73ae99.png">

Reply:
yup good catch